### ISSUE TITLE
Repo_manager assumes that samtools is in PATH

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#1064]
Depending on external executables is brittle and problematic. We should just exit with a helpful error message like "FASTA not indexed. Please run samtools faidx {}" if the FASTA file isn't indexed.

**Comment by jeromekelleher** [comment]
Also remove dependency on tabix.

**Comment by kozbo** [comment]
+1, I like this approach

**Comment by david4096** [comment]
Nice, would also close https://github.com/ga4gh/server/issues/1052 (I believe). @dcolligan what do you think about limiting the scope of the repo manager to expect indexed files already created?

**Comment by jeromekelleher** [comment]
I'd support expecting indexes to exist already for now. I think there are other directions we need to flesh repo manager out in (mostly, metadata) before we bring this functionality in.

**Comment by dcolligan** [comment]
Hm, I'm not so sure about this.  It may be true that depending on outside executables is brittle, but we can do a more involved check for them on a repo manager invocation that might use them.  My concern is that the repo manager should be as easy as possible for people to use -- the user should just be able to point it at a BAM (e.g.) file and have it work.  Creating indexes seems like something that can and should be done automatically.

**Comment by jmarshall** [comment]
Does pysam provide access to htslib's `fai_build()`?  If so, in this case you can build the _.fai_ index natively via the API you're already depending on rather than via an additional dependency on an external executable.

**Comment by macieksmuga** [comment]
The `scripts/prepare_compliance_data.py` script already does this by using equivalent samtools-equivalent `pysam` methods (and its dependencies are identical to what's required of the server code in the first place). Relevant snippet:
```
            pysam.tabix_compress(destFastaFilename, destFastaFilename + ".gz")
            refFasta = pysam.FastaFile(destFastaFilename + ".gz")
            refFasta.close()
```

**Comment by jeromekelleher** [comment]
Closed by #1166.
